docs/proposals/logging_update.md

```diff
@@ -0,0 +1,79 @@
+## Logging Updates
+
+### Problem Description
+The current state is that we are setting loggers on objects and passing loggers around. This means that each object that we create has a logger and that if we make a function, to log from that function we need to pass in the logger. The solution is to allow for logging to take place at any location in the project. This will allow developers in the future to not have to worry about where the logger is included and instead just be able to log.
+
+The broker will, instead of having  `*logger` being passed around, will instead use helper methods on a new package. This means that from now on when the broker would log something it will call the package like `log.Errorf(...)`. 
+
+The APBs will not be changed.
+
+### Work Items
+We need to decide how we are going to change this. I have put together 3 options that I think are all pretty good options that significantly help our current logging woes.
+
+1. We can keep go-logging and use package level loggers. An example is the example go program [here](https://github.com/op/go-logging/blob/master/examples/example.go). Or an example that would look more like what we would do is select a file in each package and initialize the log.  example changes to `broker.go`
```

Comment:
I kind of like the package level logging. That's actually how I did it in the unit tests by accident :)  This I think would be the least change. Add a logger to every package, then remove log from all structs. Done.